gnocchi/incoming/redis.py

```diff
@@ -113,16 +113,27 @@ def has_unprocessed(self, metric_id):
         return bool(self._client.exists(self._build_measure_path(metric_id)))
 
     @contextlib.contextmanager
-    def process_measure_for_metric(self, metric_id):
-        key = self._build_measure_path(metric_id)
-        item_len, data = self._scripts['process_measure_for_metric'](
-            keys=[key],
-        )
-
-        yield self._unserialize_measures(metric_id, data)
-
-        # ltrim is inclusive, bump 1 to remove up to and including nth item
-        self._client.ltrim(key, item_len + 1, -1)
+    def process_measure_for_metrics(self, metric_ids):
+        measures = {}
+        pipe = self._client.pipeline(transaction=False)
+        for metric_id in metric_ids:
+            key = self._build_measure_path(metric_id)
+            self._scripts['process_measure_for_metric'](
+                keys=[key],
+                client=pipe,
```

Comment:
where is client=pipe being used? it's not in script. is this some redis+lua thing?

Reply:
Not sure I understand the question. This makes the script executed by the pipe.

Reply:
if i look at process_measure_for_metric lua script above, it doesn't mention anything about client. i was just wondering if this was some param script takes in and if so you had a link for my personal knowledge.

Reply:
i guess as a followup, why dont you just build all the keys and pass it in to script? i guess you're trying to minimise change?

Reply:
Oh I get your question. The only arguments you can pass to a Lua script are `KEYS` and `ARGS`. So yeah it's a param for the pipe to execute the script (I agree the Python Redis API is not perfect in this regard).

I could pass all the keys to the script – I actually started to implement it. However it actually makes do much more Lua than Python, and I am not sure it's worth it, so I did not. It's still doable as a follow up for sure.

Reply:
ack. i'll just trust that client=pipe is something. i guess the data does seem to be attached to it eventually (assuming the gates are blindly passing)